gemini/sample-apps/swot-agent/README.md

```diff
@@ -0,0 +1,130 @@
+# swot-agent
+
+[![Python 3.10+](https://img.shields.io/badge/python-3.10+-blue.svg?logo=python&logoColor=white)](https://www.python.org/downloads/release/python-3100/)
+[![FastAPI](https://img.shields.io/badge/FastAPI-0.115.6+-green.svg)](https://fastapi.tiangolo.com)
+[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
+
+A web application that performs automated [SWOT analysis](https://en.wikipedia.org/wiki/SWOT_analysis) (Strengths, Weaknesses, Opportunities, Threats) analysis using the Google's [Gemini 2.0 model](https://ai.google.dev/gemini-api/docs/models/gemini-v2) and the [Pydantic AI](https://ai.pydantic.dev/) agent framework. The application is built with FastAPI, HTMX, and Tailwind CSS.
+
+![SWOT Analysis Demo](swot-agent.gif)
+
+The agent includes three tools:
+
+- **Content Extraction**: Extracts content from a web page given a URL
+- **Community Insights**: Calls Reddit API to get community insights from relevant subreddits
+- **Competitive Analysis**: Calls Gemini API to get competitive analysis
+
+## Getting Started
+
+### Prerequisites
+
+- Python 3.10+
+- Google Cloud API credentials
+- Optional: [Reddit API credentials](https://www.reddit.com/prefs/apps) (for Reddit content extraction)
+
+### Installation
+
+1. Clone the repository and change to the `swot-agent` directory.
+
+1. Install the required dependencies:
+
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+1. Set up your environment variables:
+
+   ```bash
+   # Google Cloud settings
+   export GOOGLE_CLOUD_PROJECT=your_project_id
+   export GOOGLE_APPLICATION_CREDENTIALS=path_to_service_account.json
+
+   # (Optional) Reddit API credentials
+   export REDDIT_CLIENT_ID=your_reddit_client_id
+   export REDDIT_CLIENT_SECRET=your_reddit_client_secret
+
+   # (Optional) Application settings
+   export APP_SECRET_KEY=your_secret_key
+   ```
+
+1. Run the application:
+
+   ```bash
+   python main.py
+   ```
+
+   You can also use the [FastAPI CLI](https://fastapi.tiangolo.com/deployment/cli/):
+
+   ```bash
+   fastapi dev --port 8080
+   ```
+
+1. Open your web browser and navigate to `http://localhost:8080`.
+
+## Usage
+
+1. Enter a valid URL in the input field
+1. Click "Analyze" to initiate the AI-powered SWOT analysis
```

Comment:
```suggestion
1. Click **"Analyze"** to initiate the AI SWOT analysis
```

Marketing wants us to avoid saying "Powered by Gemini" or "Powered by AI".